File: android_app/lib/screens/plan_selection_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../providers/app_state.dart';
import '../theme/app_theme.dart';

class PlanSelectionScreen extends StatelessWidget {
  const PlanSelectionScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Choose Your Plan'),
      ),
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(20),
          child: Column(
            children: [
              // Header
              Text(
                'Unlock Premium Features',
                style: Theme.of(context).textTheme.displayMedium,
                textAlign: TextAlign.center,
              ),
              const SizedBox(height: 8),
              Text(
                'Scale your YouTube automation with Pro',
                style: Theme.of(context).textTheme.bodyMedium?.copyWith(
                  color: AppTheme.textSecondary,
                ),
                textAlign: TextAlign.center,
              ),
              
              const SizedBox(height: 32),
              
              // Free Plan Card
              Consumer<AppState>(
                builder: (context, appState, _) {
                  final isCurrent = appState.currentUser?.isFree ?? true;
                  
                  return _PlanCard(
                    name: 'Free',
                    price: '\$0',
                    period: 'forever',
                    isCurrent: isCurrent,
                    features: const [
                      '6 videos per month',
                      '3 active schedulers',
                      'Bring your own Suno API key',
                      'Standard video quality',
                      'Community support',
                    ],
                    onSelect: () {
                      if (!isCurrent) {
                        // Downgrade logic
                        ScaffoldMessenger.of(context).showSnackBar(
                          const SnackBar(
                            content: Text('Contact support to downgrade'),
                          ),
                        );
                      }
                    },
                  );
                },
              ),
              
              const SizedBox(height: 16),
              
              // Pro Plan Card
              Consumer<AppState>(
                builder: (context, appState, _) {
                  final isCurrent = appState.currentUser?.isPro ?? false;
                  
                  return _PlanCard(
                    name: 'Pro',
                    price: '\$29',
                    period: 'per month',
                    isCurrent: isCurrent,
                    isPro: true,
                    features: const [
                      'Unlimited videos',
                      'Unlimited schedulers',
                      'Suno API included',
                      'HD video quality',
                      'Priority support',
                      'Advanced analytics',
                      'Custom branding',
                    ],
                    badge: 'POPULAR',
                    onSelect: () {
                      if (!isCurrent) {
                        _showUpgradeDialog(context);
                      }
                    },
                  );
                },
              ),
              
              const SizedBox(height: 24),
              
              // FAQ
              Text(
                'Frequently Asked Questions',
                style: Theme.of(context).textTheme.titleMedium,
              ),
              const SizedBox(height: 16),
              
              _FAQItem(
                question: 'Can I cancel anytime?',
                answer: 'Yes, you can cancel your Pro subscription at any time. You\'ll have access until the end of your billing period.',
              ),
              const SizedBox(height: 12),
              _FAQItem(
                question: 'What is a Suno API key?',
                answer: 'Suno is the AI music generation service we use. Free users provide their own API key, while Pro users get unlimited access included.',
              ),
              const SizedBox(height: 12),
              _FAQItem(
                question: 'Can I upgrade later?',
                answer: 'Absolutely! You can upgrade to Pro at any time from the Settings screen.',
              ),
            ],
          ),
        ),
      ),
    );
  }

  void _showUpgradeDialog(BuildContext context) {
    showDialog(
      context: context,
      builder: (context) => AlertDialog(
        title: const Text('Upgrade to Pro'),
        content: const Text('Pro plan upgrades will be available soon via Stripe integration.'),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(context),
            child: const Text('OK'),
          ),
        ],
      ),
    );
  }
}

// Plan Card Widget
class _PlanCard extends StatelessWidget {
  final String name;
  final String price;
  final String period;
  final bool isCurrent;
  final bool isPro;
  final List<String> features;
  final String? badge;
  final VoidCallback onSelect;

  const _PlanCard({
    required this.name,
    required this.price,
    required this.period,
    required this.isCurrent,
    this.isPro = false,
    required this.features,
    this.badge,
    required this.onSelect,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.all(24),
      decoration: BoxDecoration(
        color: isPro ? AppTheme.primaryColor.withOpacity(0.1) : null,
        borderRadius: BorderRadius.circular(16),
        border: Border.all(
          color: isPro
              ? AppTheme.primaryColor
              : Colors.white.withOpacity(0.1),
          width: isPro ? 2 : 1,
        ),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Text(
                name,
                style: Theme.of(context).textTheme.titleLarge,
              ),
              if (badge != null)
                Container(
                  padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
                  decoration: BoxDecoration(
                    color: AppTheme.primaryColor,
                    borderRadius: BorderRadius.circular(8),
                  ),
                  child: Text(
                    badge!,
                    style: Theme.of(context).textTheme.bodySmall?.copyWith(
                      color: Colors.white,
                      fontWeight: FontWeight.w700,
                    ),
                  ),
                ),
            ],
          ),
          const SizedBox(height: 12),
          Row(
            crossAxisAlignment: CrossAxisAlignment.end,
            children: [
              Text(
                price,
                style: Theme.of(context).textTheme.displayMedium?.copyWith(
                  color: isPro ? AppTheme.primaryColor : null,
                ),
              ),
              const SizedBox(width: 8),
              Padding(
                padding: const EdgeInsets.only(bottom: 6),
                child: Text(
                  period,
                  style: Theme.of(context).textTheme.bodyMedium?.copyWith(
                    color: AppTheme.textSecondary,
                  ),
                ),
              ),
            ],
          ),
          const SizedBox(height: 20),
          ...features.map((feature) {
            return Padding(
              padding: const EdgeInsets.only(bottom: 12),
              child: Row(
                children: [
                  Icon(
                    Icons.check_circle,
                    size: 20,
                    color: isPro ? AppTheme.primaryColor : AppTheme.success,
                  ),
                  const SizedBox(width: 12),
                  Expanded(
                    child: Text(
                      feature,
                      style: Theme.of(context).textTheme.bodyMedium,
                    ),
                  ),
                ],
              ),
            );
          }),
          const SizedBox(height: 16),
          SizedBox(
            width: double.infinity,
            child: ElevatedButton(
              onPressed: isCurrent ? null : onSelect,
              style: ElevatedButton.styleFrom(
                backgroundColor: isPro ? AppTheme.primaryColor : AppTheme.surfaceDark,
              ),
              child: Text(isCurrent ? 'Current Plan' : 'Select Plan'),
            ),
          ),
        ],
      ),
    );
  }
}

// FAQ Item Widget
class _FAQItem extends StatelessWidget {
  final String question;
  final String answer;

  const _FAQItem({
    required this.question,
    required this.answer,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.all(16),
      decoration: AppTheme.cardDecoration(),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            question,
            style: Theme.of(context).textTheme.titleSmall,
          ),
          const SizedBox(height: 8),
          Text(
            answer,
            style: Theme.of(context).textTheme.bodyMedium?.copyWith(
              color: AppTheme.textSecondary,
            ),
          ),
        ],
      ),
    );
  }
}

File: android_app/app/src/main/java/com/autosong/studio/ui/screens/PlanSelectionScreen.kt
package com.autosong.studio.ui.screens

import androidx.compose.foundation.*
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import com.autosong.studio.ui.theme.*

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun PlanSelectionScreen(
    navController: NavController
) {
    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Choose Your Plan") },
                navigationIcon = {
                    IconButton(onClick = { navController.popBackStack() }) {
                        Icon(Icons.Default.ArrowBack, contentDescription = "Back")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = BackgroundDark)
            )
        }
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .verticalScroll(rememberScrollState())
                .padding(20.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Free Plan
            PlanCard(
                title = "Free",
                price = "$0",
                period = "/month",
                features = listOf(
                    "1 Scheduler",
                    "10 Videos/month",
                    "Bring your own Suno API key",
                    "Basic support"
                ),
                isPro = false,
                onSelect = { navController.popBackStack() }
            )

            // Pro Plan
            PlanCard(
                title = "Pro",
                price = "$19",
                period = "/month",
                features = listOf(
                    "Unlimited Schedulers",
                    "Unlimited Videos",
                    "Included Suno API credits",
                    "Priority support",
                    "Advanced analytics"
                ),
                isPro = true,
                onSelect = { /* Payment flow */ }
            )

            Text(
                "Plans can be changed anytime. Cancel at any time.",
                style = MaterialTheme.typography.bodySmall,
                color = TextSecondary,
                textAlign = TextAlign.Center,
                modifier = Modifier.fillMaxWidth()
            )
        }
    }
}

@Composable
private fun PlanCard(
    title: String,
    price: String,
    period: String,
    features: List<String>,
    isPro: Boolean,
    onSelect: () -> Unit
) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = if (isPro) PrimaryColor.copy(alpha = 0.1f) else SurfaceDark
        ),
        border = BorderStroke(
            2.dp,
            if (isPro) PrimaryColor else Color.White.copy(alpha = 0.05f)
        )
    ) {
        Column(modifier = Modifier.padding(20.dp)) {
            if (isPro) {
                Surface(
                    color = PrimaryColor,
                    shape = RoundedCornerShape(4.dp)
                ) {
                    Text(
                        "RECOMMENDED",
                        modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp),
                        style = MaterialTheme.typography.labelSmall,
                        fontWeight = FontWeight.Bold
                    )
                }
                Spacer(modifier = Modifier.height(12.dp))
            }

            Text(title, style = MaterialTheme.typography.titleLarge, fontWeight = FontWeight.Bold)
            
            Row(verticalAlignment = Alignment.Bottom) {
                Text(price, style = MaterialTheme.typography.displayMedium, fontWeight = FontWeight.ExtraBold)
                Text(period, style = MaterialTheme.typography.bodyMedium, color = TextSecondary)
            }

            Spacer(modifier = Modifier.height(16.dp))

            features.forEach { feature ->
                Row(
                    modifier = Modifier.padding(vertical = 4.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        Icons.Default.Check,
                        contentDescription = null,
                        tint = if (isPro) PrimaryColor else Success,
                        modifier = Modifier.size(20.dp)
                    )
                    Spacer(modifier = Modifier.width(12.dp))
                    Text(feature, style = MaterialTheme.typography.bodyMedium)
                }
            }

            Spacer(modifier = Modifier.height(16.dp))

            Button(
                onClick = onSelect,
                modifier = Modifier.fillMaxWidth(),
                colors = ButtonDefaults.buttonColors(
                    containerColor = if (isPro) PrimaryColor else SurfaceHighlight
                )
            ) {
                Text(if (isPro) "Upgrade to Pro" else "Continue with Free")
            }
        }
    }
}
